### PR TITLE
fix(synchronize): refactor synchronize function

### LINKED ISSUE #69: 🚀 | New scenarios must be created by synchronize function
## Describe your use case

- create a new local scenario.
- call synchronize function with updateRemote parameter to true.
- scenario json file is created locally but there is no new scenario created on remote server.

## Describe the solution you'd like
New scenarios must be created by synchronize function

### PR DESCRIPTION
**Pull request by KarimGl**
#### Issue Number
fixes #69 #68 #71 

#### Describe the changes you've made
- refactor synchronize function
- synchronize function can create new scenario
- update remote scenario using scenario raw ws intead of executing sql query
- add explanatory errors

#### Test plan 
Run ScenarioSynchronizeServiceTest

#### Checklist

- [x] Refer to issue(s) the PR solves
- [x] New kotlin code is covered by tests
- [ ] Add screenshots or gifs of the new behavior, if applicable.
- [x] All new and existing tests pass
- [x] No git conflict

### CODE REVIEW

**Review by nbrouand — COMMENTED** [review]
ok but could you refactor synchronize method in order to be more readable pls

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/synchronize/ChutneyServerServiceImpl.kt [hunk 1]
```diff
@@ -29,12 +37,38 @@ object ChutneyServerServiceImpl : ChutneyServerService {
...
 
-    override fun updateJsonScenario(serverInfo: ChutneyServerInfo, scenarioContent: String, scenarioId: String) {
+    override fun updateJsonScenario(serverInfo: ChutneyServerInfo, scenario: ChutneyScenario) {
+        var tags: List<String> = getScenarioTags(serverInfo, scenario)
+        val generatedTag = "GENERATED"
+        if (!tags.contains(generatedTag)) {
+            tags = tags.plus(generatedTag);
+        }
+        val body =
+            "{\"id\": ${SELENIUM_BY.id} , \"content\":\"${StringEscapeUtils.escapeJson(scenario.toString())}\", \"title\": \"${scenario.title}\", \"description\":\"${scenario.description}\" , \"tags\": ${
```
Comment (boddissattva) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/synchronize/ChutneyServerServiceImpl.kt:47]: Maybe use here [Kotlin raw string](https://kotlinlang.org/docs/strings.html#raw-strings) to have a cleaner JSON.
Same for the **createJsonScenario()** funtion.

File: dsl/src/test/kotlin/com/chutneytesting/kotlin/synchronize/ScenarioSynchronizeServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        assertScenarioSynchronization(
+            tempDir = tempDir,
+            scenario = modifiedScenario,
+            id = createdScenarioId,
+            updateRemote = true
+        )
+        val createScenarioRequest = HttpRequest().withPath("/api/scenario/v2/raw").withMethod("POST")
+        mockServer.verify(createScenarioRequest)
+        val requestJson = mockServer.retrieveRecordedRequests(createScenarioRequest, Format.JSON)
+        Assertions.assertThat(requestJson).contains(localScenario.title)
```
Comment (boddissattva) [inline dsl/src/test/kotlin/com/chutneytesting/kotlin/synchronize/ScenarioSynchronizeServiceTest.kt:94]: Could be statically imported.

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/synchronize/ScenarioSynchronizeService.kt [hunk 3]
```diff
@@ -13,20 +15,64 @@ fun ChutneyScenario.synchronise(
...
+
+
+private fun renameJsonFile(
+    jsonFile: File,
+    fileName: String
+) {
+    try {
+        Files.move(jsonFile.toPath(), jsonFile.toPath().resolveSibling(fileName))
+    } catch (e: IOException) {
+        println("| AT json file at ${jsonFile.name} cannot be renamed to: $fileName")
```
Comment (nbrouand) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/synchronize/ScenarioSynchronizeService.kt:75]: Add exception message

**Review by nbrouand — APPROVED** [review]

**Pull request merged**